<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ImgProc_OpencvClass</class>
 <widget class="QMainWindow" name="ImgProc_OpencvClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1364</width>
    <height>940</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ImgProc_Opencv</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>430</y>
      <width>1191</width>
      <height>401</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(255, 255, 255);</string>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab_ImgProcBasicAlg">
     <attribute name="title">
      <string>图像处理基本算法</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>930</x>
        <y>30</y>
        <width>131</width>
        <height>91</height>
       </rect>
      </property>
      <property name="title">
       <string>图像处理基本操作</string>
      </property>
      <widget class="QRadioButton" name="radioButton_ToGrayImg">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>61</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>灰度图</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioButton_ToBlackWhiteImg">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>61</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>二值化</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="geometry">
       <rect>
        <x>1090</x>
        <y>50</y>
        <width>71</width>
        <height>81</height>
       </rect>
      </property>
      <property name="title">
       <string>颜色空间</string>
      </property>
      <widget class="QRadioButton" name="radioButton_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>61</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>RGB</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioButton_6">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>61</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>HSV</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_6">
      <property name="geometry">
       <rect>
        <x>930</x>
        <y>640</y>
        <width>111</width>
        <height>81</height>
       </rect>
      </property>
      <property name="title">
       <string>直方图</string>
      </property>
      <widget class="QRadioButton" name="radioButton_13">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>91</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>灰度直方图</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioButton_14">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>81</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>RGB直方图</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_10">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>20</y>
        <width>671</width>
        <height>291</height>
       </rect>
      </property>
      <property name="title">
       <string>2.imgpro组件学习</string>
      </property>
      <widget class="QGroupBox" name="groupBox_7">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>381</width>
         <height>261</height>
        </rect>
       </property>
       <property name="title">
        <string>图像处理</string>
       </property>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>81</width>
          <height>91</height>
         </rect>
        </property>
        <property name="title">
         <string>线性滤波</string>
        </property>
        <widget class="QLabel" name="label_7">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>20</y>
           <width>71</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>1.方框滤波</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_8">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>40</y>
           <width>71</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>2.平均滤波</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_9">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>60</y>
           <width>71</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>3.高斯滤波</string>
         </property>
        </widget>
       </widget>
       <widget class="QGroupBox" name="groupBox_5">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>20</y>
          <width>81</width>
          <height>91</height>
         </rect>
        </property>
        <property name="title">
         <string>非线性滤波</string>
        </property>
        <widget class="QLabel" name="label_10">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>20</y>
           <width>71</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>4.中值滤波</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_11">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>50</y>
           <width>71</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>5.双边滤波</string>
         </property>
        </widget>
       </widget>
       <widget class="QGroupBox" name="groupBox_8">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>140</y>
          <width>191</width>
          <height>91</height>
         </rect>
        </property>
        <property name="title">
         <string>形态学滤波</string>
        </property>
        <widget class="QLabel" name="label_12">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>20</y>
           <width>41</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>1.膨胀</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_13">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>40</y>
           <width>41</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>2.腐蚀</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_14">
         <property name="geometry">
          <rect>
           <x>60</x>
           <y>20</y>
           <width>54</width>
           <height>12</height>
          </rect>
         </property>
         <property name="text">
          <string>3.开运算</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_15">
         <property name="geometry">
          <rect>
           <x>60</x>
           <y>40</y>
           <width>54</width>
           <height>12</height>
          </rect>
         </property>
         <property name="text">
          <string>4.闭运算</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_16">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>20</y>
           <width>54</width>
           <height>12</height>
          </rect>
         </property>
         <property name="text">
          <string>5.顶帽</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_17">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>40</y>
           <width>54</width>
           <height>12</height>
          </rect>
         </property>
         <property name="text">
          <string>6.黑帽</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_18">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>60</y>
           <width>81</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>7.形态学梯度</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="radioButton_Morphology">
         <property name="geometry">
          <rect>
           <x>100</x>
           <y>60</y>
           <width>71</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>综合示例</string>
         </property>
        </widget>
       </widget>
       <widget class="QRadioButton" name="radioButton_16">
        <property name="geometry">
         <rect>
          <x>210</x>
          <y>30</y>
          <width>71</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>漫水填充</string>
        </property>
       </widget>
       <widget class="QRadioButton" name="radioButton_20">
        <property name="geometry">
         <rect>
          <x>210</x>
          <y>90</y>
          <width>171</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>图像金字塔与图片尺寸缩放</string>
        </property>
       </widget>
       <widget class="QRadioButton" name="radioButton_23">
        <property name="geometry">
         <rect>
          <x>210</x>
          <y>140</y>
          <width>71</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>阈值化</string>
        </property>
       </widget>
       <widget class="QRadioButton" name="radioButton_Filter">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>110</y>
          <width>121</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>五种滤波综合示例</string>
        </property>
       </widget>
      </widget>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="geometry">
        <rect>
         <x>410</x>
         <y>30</y>
         <width>241</width>
         <height>161</height>
        </rect>
       </property>
       <property name="title">
        <string>图像变换</string>
       </property>
       <widget class="QGroupBox" name="groupBox_11">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>91</width>
          <height>131</height>
         </rect>
        </property>
        <property name="title">
         <string>边缘检测</string>
        </property>
        <widget class="QLabel" name="label_3">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>20</y>
           <width>54</width>
           <height>12</height>
          </rect>
         </property>
         <property name="text">
          <string>1.canny</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_4">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>40</y>
           <width>54</width>
           <height>12</height>
          </rect>
         </property>
         <property name="text">
          <string>2.sobel</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_5">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>60</y>
           <width>71</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>3.Laplacian</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_6">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>80</y>
           <width>51</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>4.scharr</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="radioButton_EdgeDetection">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>100</y>
           <width>81</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>综合示例</string>
         </property>
        </widget>
       </widget>
       <widget class="QRadioButton" name="radioButton_36">
        <property name="geometry">
         <rect>
          <x>130</x>
          <y>50</y>
          <width>61</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>重映射</string>
        </property>
       </widget>
       <widget class="QRadioButton" name="radioButton_37">
        <property name="geometry">
         <rect>
          <x>130</x>
          <y>20</y>
          <width>71</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>霍夫变换</string>
        </property>
       </widget>
       <widget class="QRadioButton" name="radioButton_38">
        <property name="geometry">
         <rect>
          <x>130</x>
          <y>80</y>
          <width>81</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>仿射变换</string>
        </property>
       </widget>
       <widget class="QRadioButton" name="radioButton_39">
        <property name="geometry">
         <rect>
          <x>130</x>
          <y>110</y>
          <width>91</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>直方图均衡化</string>
        </property>
       </widget>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_9">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>193</width>
        <height>311</height>
       </rect>
      </property>
      <property name="title">
       <string>1.core组件学习</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QRadioButton" name="radioButton_24">
         <property name="text">
          <string>基本数据结构学习使用</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QRadioButton" name="radioButton_29">
         <property name="text">
          <string>访问图像中像素的三种方法</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QRadioButton" name="radioButton_25">
         <property name="text">
          <string>ROI区域图像叠加和图像混合</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QRadioButton" name="radioButton_26">
         <property name="text">
          <string>对比度和亮度值调整</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QRadioButton" name="radioButton_27">
         <property name="text">
          <string>傅里叶变换</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QRadioButton" name="radioButton_28">
         <property name="text">
          <string>XML和YAML文件应用</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_ImgProcComplexAlg">
     <attribute name="title">
      <string>图像处理复杂算法</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox_12">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>0</y>
        <width>291</width>
        <height>291</height>
       </rect>
      </property>
      <property name="title">
       <string>模板匹配</string>
      </property>
      <widget class="QPushButton" name="pushButton">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>100</y>
         <width>61</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>打开模板</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_19">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>141</width>
         <height>121</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: rgb(0, 0, 0);</string>
       </property>
       <property name="text">
        <string>TextLabel</string>
       </property>
      </widget>
      <widget class="QComboBox" name="comboBox">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>50</y>
         <width>111</width>
         <height>22</height>
        </rect>
       </property>
       <item>
        <property name="text">
         <string>平方差匹配法</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>相关匹配法</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>相关系数匹配法</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>归一化平方差匹配法</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>归一化相关匹配法</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>归一化相关系数匹配法</string>
        </property>
       </item>
      </widget>
      <widget class="QLabel" name="label_20">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>41</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>模板：</string>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="QWidget" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>10</y>
      <width>1251</width>
      <height>411</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(255, 255, 255);</string>
    </property>
    <widget class="QLabel" name="label_ProcImg">
     <property name="geometry">
      <rect>
       <x>590</x>
       <y>30</y>
       <width>541</width>
       <height>361</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(0, 0, 0);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_ShowImg">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>561</width>
       <height>361</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(0, 255, 255);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>54</width>
       <height>12</height>
      </rect>
     </property>
     <property name="text">
      <string>原图像</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>590</x>
       <y>10</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>处理后图像</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_OpenCamera">
     <property name="geometry">
      <rect>
       <x>1160</x>
       <y>40</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>打开相机</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_SnapShot">
     <property name="geometry">
      <rect>
       <x>1160</x>
       <y>100</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>拍照</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1364</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="ImgProc_Opencv.qrc"/>
 </resources>
 <connections/>
</ui>
